keps/sig-node/xxx-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1408 @@
+  are provided by the same resource plugin, which is identified by a
+  field in the class.
+- ResourceClaim, namespaced, with parameters provided by a normal user
+  that describes a resource instance that needs to be allocated. A
+  ResourceClaim contains the usual meta data, a spec and a status. The
+  spec identifies the plugin that handles the resource via a class
+  name.
+
+To support arbitrarily complex parameters, both ResourceClass and
+ResourceClaim contain one field which holds a
+runtime.RawExtension. Validation can be handled by plugins through an
+admission controller (if desired) or later at runtime when the
+parameters are passed to the plugin.
+
+The ResourceClaim spec is read-only once created. The ResourceClaim
+status is reserved for system usage and holds the current state of the
+resource. The status must not get lost. This is departing from how
+Kubernetes traditionally handled status, but something that more
+recently [became more
+acceptable](https://github.com/kubernetes/enhancements/pull/2537). Kube-scheduler
+and plugin communicate by modifying that status. The status is also
+how Kubernetes tracks that a plugin has allocated the resource and on
+which nodes the resource is available.
+
+This approach is an intentional simplification compared to the PV/PVC
+model for volumes because we don't need to deal with two objects when
+allocating resources and therefore don't need something like the
+volume binding controller. If desired, a resource plugin can implement
+support for manually allocated (“static provisioning” in the context
+of volumes) and/or existing resources by reacting to ResourceClaims by
+using those resources to satisfy a claim.
+
+Allocation of a resource happens either immediately (“immediate
+allocation”) or when a Pod needs the resource (“delayed allocation”),
+depending on a flag in the ResourceClaim spec. Pods reference resource
+claims in a new PodSpec.Resources list. Each resource in that list
+then can be made available to one or more containers in that Pod.  To
+support ephemeral resources, an entry in the new PodSpec.Resources
+list can also be a ResourceClaimTemplate. When a Pod gets created,
+such a template will be used to create a normal ResourceClaim with the
+Pod as owner, and then the normal allocation of the resource takes
+place.
+
+For immediate allocation, scheduling Pods is simple because the
+resource is already allocated and determines the nodes on which the
+Pod may run. For delayed allocation, a node is selected tentatively
+and plugin(s) try to allocate their resources for that node. If that
+succeeds, the Pod can start to run. If it fails, the scheduler must
+determine whether some other node fits the requirements and if so,
+request allocation again. If no node fits because some resources were
+already allocated for a node and are only usable there, then those
+resources must be released and then get allocated elsewhere.
```

Comment:
So is the proposal to (somehow) do away with the kubelet's internal device manager entirely, and have all device management happen decentralized in the plugins themselves? How would this work to coordinate with the `TopologyManager` (or some future iteration of something like this) which needs to align resources in some way (e.g. on a common NUMA node, for example)?